pkg/prometheus/resource_selector.go

```diff
@@ -830,13 +840,38 @@ func (rs *ResourceSelector) SelectScrapeConfigs(ctx context.Context, listFn List
 	return res, nil
 }
 
+// ValidateSelectorRoles validate the roles based on main role.
+func ValidateSelectorRoles(c monitoringv1alpha1.KubernetesSDConfig) error {
+
+	foundSelectorRoles := make(map[monitoringv1alpha1.Role]struct{})
+	allowedSelectors := map[monitoringv1alpha1.Role][]string{
+		RolePod:           {string(RolePod)},
+		RoleService:       {string(RoleService)},
+		RoleEndpointSlice: {string(RolePod), string(RoleService), string(RoleEndpointSlice)},
+		RoleEndpoint:      {string(RolePod), string(RoleService), string(RoleEndpoint)},
+		RoleNode:          {string(RoleNode)},
+		RoleIngress:       {string(RoleIngress)},
+	}
+
+	for _, selector := range c.Selectors {
+
+		if _, ok := foundSelectorRoles[selector.Role]; ok {
```

Comment:
We need to apply strings.ToLower() on the role value (e.g. prometheus-operator both "Pod" and "pod").